utils/metric_database.py

```diff
@@ -0,0 +1,295 @@
+"""Postgres metric database utilities."""
+import logging
+import os
+from dataclasses import dataclass
+from typing import Any, Dict, List, Optional, Tuple, Union, cast
+
+from psycopg2 import Error, extensions, pool
+
+logging.basicConfig(
+    level=logging.INFO,
+    format="%(asctime)s [%(levelname)s] %(message)s",
+    handlers=[logging.StreamHandler()],
+)
+
+
+@dataclass
+class DatabaseCredentials:
+    """Class for getting and validating env variables.
+
+    Raises:
+        ValueError: if any of the environment variables is not set or empty, then we can connect to the database
+    """
+
+    def __init__(self) -> None:
+        """Constructor for DatabaseCredentials to get the required environment variables."""
+        self.db_name = os.getenv("DB_NAME")
+        self.db_host = os.getenv("DB_HOST")
+        self.db_user = os.getenv("DB_USER")
+        self.db_password = os.getenv("DB_PASSWORD")
+        self.db_port = os.getenv("DB_PORT")
+
+    def validate_env(self) -> None:
+        """Checks whether the 5 environment variables required to connect to database is empty.
+
+        Raises:
+            ValueError: if any one of the 5 environment variables is empty or not set
+        """
+        for var, value in vars(self).items():
+            if value is None or len(value) == 0:
+                raise ValueError(f"{var.upper()} is not set")
+
+
+class SQLQueries:
+    """Class containing queries for creating and checking relation existence."""
+
+    @staticmethod
+    def create_readability_relation_query() -> str:
+        """SQL query for creating the readability relation with 3 columns, ID, TimeStamp and ReadabilityScore.
+
+        Returns:
+            str: SQL query for creating the readability relation
+        """
+        sql_query = """
+            CREATE TABLE "readability" (
+                "id" SERIAL PRIMARY KEY,
+                "time_stamp" TIMESTAMP,
+                "readability_score" FLOAT(2)
```

Comment:
Do we want to add what response text was as well in the table? Having only timestamp and score does not provide any clue as to for what score is for which response text.

Reply:
We did think about adding those, but Jon and I we don't want to be storing user data for now.